icu4j/main/tests/core/src/com/ibm/icu/dev/test/format/MeasureUnitTest.java

```diff
@@ -3923,12 +3923,11 @@ public void TestIdentifiers() {
             new TestCase("exbibyte", "exbibyte"),
             new TestCase("zebibyte", "zebibyte"),
             new TestCase("yobibyte", "yobibyte"),
-
             // Testing sort order of prefixes.
-            //
-            // TODO(icu-units#70): revisit when fixing normalization. For now we're
-            // just checking some consistency between C&J.
-            new TestCase("megafoot-mebifoot-kibifoot-kilofoot", "kibifoot-mebifoot-kilofoot-megafoot"),
+            new TestCase("megafoot-mebifoot-kibifoot-kilofoot", "mebifoot-megafoot-kibifoot-kilofoot"),
+            // Testing sort of units.
+            new TestCase("inch-foot-yard", "foot-inch-yard"), // TODO: the result should be "yard-foot-inch"
```

Comment:
As with C++, the TODO doesn't look spec-compliant to me, I think the sort order is undefined.

Also: add negative exponents tests (e.g. milli, micro, nano).

Reply:
done